simulator/extender/httpextender.go

```diff
@@ -0,0 +1,190 @@
+package extender
+
+//go:generate mockgen --build_flags=--mod=mod -destination=./mock_$GOPACKAGE/httpextender.go . HTTPExtender
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/sets"
+	restclient "k8s.io/client-go/rest"
+	extenderv1 "k8s.io/kube-scheduler/extender/v1"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+const (
+	// DefaultExtenderTimeout defines the default extender timeout in second.
+	DefaultExtenderTimeout = 5 * time.Second
+)
+
+type HTTPExtender interface {
```

Comment:
```suggestion
// HTTPExtender calls the actual Extender configured through KubeSchedulerConfig by users.
type HTTPExtender interface {
```

Reply:
It's kinda hard to know the meaning from the name `HTTPExtender`.
`ExtenderCaller` might be better.